### PR TITLE
Move examples to fn api

### PR DESCRIPTION
**Pull request by JanuszL**
- move all use cases to the functional API
- reworks SSD, RN50, super resolution and TSM examples

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It moves network examples to the functional API

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     reworks SSD, RN50, super resolution and TSM examples to use the functional API
 - Affected modules and functionalities:
     use_cases part of the documentation
 - Key points relevant for the review:
     functional API usage
 - Validation and testing:
     CI
 - Documentation (including examples):
     documentation and examples have been updated

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Review by mzient — APPROVED** [review]

File: docs/examples/use_cases/paddle/resnet50/main.py [hunk 1]
```diff
@@ -202,14 +166,32 @@ def main():
     FLAGS.device_id = int(env['FLAGS_selected_gpus'])
     FLAGS.whole_batch_size = FLAGS.world_size * FLAGS.batch_size
 
-    pipe = HybridTrainPipe()
+    pipe = create_dali_pipeline(batch_size=FLAGS.batch_size,
+                             num_threads=FLAGS.num_threads,
```
Comment (jantonguirao) [inline docs/examples/use_cases/paddle/resnet50/main.py:170]: nitpick: indent

File: docs/examples/use_cases/paddle/resnet50/main.py [hunk 2]
```diff
@@ -202,14 +166,32 @@ def main():
...
     pipe.build()
-    sample_per_shard = len(pipe) // FLAGS.world_size
+    sample_per_shard = pipe.epoch_size("Reader") // FLAGS.world_size
     train_loader = DALIClassificationIterator(pipe, reader_name="Reader",
                                               last_batch_policy=LastBatchPolicy.PARTIAL)
 
     if FLAGS.local_rank == 0:
-        pipe = HybridValPipe()
+        pipe = create_dali_pipeline(batch_size=FLAGS.batch_size,
+                                num_threads=FLAGS.num_threads,
```
Comment (jantonguirao) [inline docs/examples/use_cases/paddle/resnet50/main.py:186]: nitpick: indent

File: docs/examples/use_cases/paddle/ssd/train.py [hunk 3]
```diff
@@ -23,94 +23,75 @@
...
+                                                name="Reader")
+
+        crop_begin, crop_size, bboxes, labels = fn.random_bbox_crop(bboxes, labels,
+                                                                    device="cpu",
+                                                                    aspect_ratio=[0.5, 2.0],
+                                                                    thresholds=[0, 0.1, 0.3, 0.5, 0.7, 0.9],
+                                                                    scaling=[0.3, 1.0],
+                                                                    bbox_layout="xyXY",
+                                                                    allow_no_crop=True,
+                                                                    num_attempts=1)
```
Comment (jantonguirao) [inline docs/examples/use_cases/paddle/ssd/train.py:64]: you changed from 50 attempts to 1. Why is that?

File: docs/examples/use_cases/paddle/ssd/train.py [hunk 4]
```diff
@@ -23,94 +23,75 @@
...
+        images = fn.image_decoder_slice(
+            images, crop_begin, crop_size, device="cpu", output_type=types.RGB)
```
Comment (jantonguirao) [inline docs/examples/use_cases/paddle/ssd/train.py:66]: before there was ```device='mixed'```. Did you change it in purpose?

File: docs/examples/use_cases/pytorch/single_stage_detector/src/coco_pipeline.py [hunk 5]
```diff
@@ -15,110 +15,84 @@
...
+                                                                    device="cpu",
+                                                                    aspect_ratio=[
+                                                                        0.5, 2.0],
+                                                                    thresholds=[
+                                                                        0, 0.1, 0.3, 0.5, 0.7, 0.9],
+                                                                    scaling=[
+                                                                        0.3, 1.0],
+                                                                    bbox_layout="xyXY",
+                                                                    allow_no_crop=True,
+                                                                    num_attempts=1)
```
Comment (jantonguirao) [inline docs/examples/use_cases/pytorch/single_stage_detector/src/coco_pipeline.py:53]: I wonder if the number of attempts being 1 was intentional in the first place.

File: docs/examples/use_cases/video_superres/dataloading/dataloaders.py [hunk 6]
```diff
@@ -11,44 +11,47 @@
...
+    pipeline = Pipeline(batch_size, num_threads, device_id, seed=12)
+    with pipeline:
+        images = fn.video_reader(device="gpu", filenames=files, sequence_length=sequence_length,
+                                 normalized=False, random_shuffle=True, image_type=types.RGB,
+                                 dtype=types.UINT8, initial_fill=16, pad_last_batch=True, name="Reader")
+        images = ops.crop(images, crop=crop_size, dtype=types.FLOAT,
+                          crop_pos_x=fn.uniform(range=(0.0, 1.0)),
+                          crop_pos_y=fn.uniform(range=(0.0, 1.0)))
+
+        images = fn.tanspose(images, perm=[3, 0, 1, 2])
```
Comment (jantonguirao) [inline docs/examples/use_cases/video_superres/dataloading/dataloaders.py:28]:
```suggestion
        images = fn.transpose(images, perm=[3, 0, 1, 2])
```

Comment (JanuszL) [reply docs/examples/use_cases/video_superres/dataloading/dataloaders.py:28]: Done

File: docs/examples/use_cases/pytorch/single_stage_detector/src/coco_pipeline.py (hunk 5 above)
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/single_stage_detector/src/coco_pipeline.py:53]: Copy-paste. Done

File: docs/examples/use_cases/paddle/ssd/train.py (hunk 4 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/ssd/train.py:66]: Done

File: docs/examples/use_cases/paddle/ssd/train.py (hunk 3 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/ssd/train.py]: Copy-paste. Done

File: docs/examples/use_cases/paddle/resnet50/main.py (hunk 2 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/main.py]: Done

File: docs/examples/use_cases/paddle/resnet50/main.py (hunk 1 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/main.py]: Done

**Review by jantonguirao — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Pull request merged**